### PR TITLE
General config field

### PR DESCRIPTION
**Pull request by dchen150**
Config that supports string, numbers, and booleans to be added, modified, and deleted from firestore

screenrecording-chunk-1

screenrecording-chunk-2

screenrecording-chunk-3

screenrecording-chunk-4

### CODE REVIEW

**Comment by dchen150** [comment]
@kozr lmk if this is what you had in mind for the general config 😁

**Review by kozr — CHANGES_REQUESTED** [review]
Thanks @dchen150, I totally missed this PR. I'll take a careful look at the code tonight but from first glance, it seems like you've committed package-lock.json as well.

**Comment by dchen150** [comment]
@kozr ya oops my bad. do you know how I can undo that 😬

File: components/generalConfig.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        : `BuildConfig.componentStyling.${title}`
+    }.${field.field}`;
+
+    let fieldValue = field.value;
+
+    if (modalAction === DELETE) {
+      fieldValue = firestore.FieldValue.delete();
+    } else if (field.type === 'number') {
+      // convert to corresponding type if necessary
+      fieldValue = Number(fieldValue);
```
Comment (kozr) [inline components/generalConfig.js:118]: oh I didn't know `Number()` was a thing. Cool

File: components/generalConfig.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      if (!fieldValue) {
+        alert('Field must be a valid number.');
+        return;
+      }
+    } else if (field.type === 'boolean') {
+      fieldValue = fieldValue === 'true';
```
Comment (kozr) [inline components/generalConfig.js:124]:
```suggestion
      fieldValue = (fieldValue === 'true');
```
Just to make it prettier

**Comment by kozr** [comment]
I like your code!

**Comment by kozr** [comment]
> @kozr ya oops my bad. do you know how I can undo that 😬

I think you can just do `git rm package-lock.json` and commit

**Comment by kozr** [comment]
Although I am seeing some error codes despite you having a key attribute. Is this from the previous PR?
image

**Comment by dchen150** [comment]
@kozr  thanks for reviewing and fixing up the key issue - i think it was left behind from a previous PR. I removed package.json

File: components/generalConfig.js (hunk 2 above)
Comment (dchen150) [reply components/generalConfig.js]: image
it goes against the prettier linter LOL

**Review by kozr — APPROVED** [review]

**Pull request merged**